### PR TITLE
[UNI-323] feat : Stream + SSE로 대용량 조회 분할처리 로직 구현

### PR DESCRIPTION
**Pull request by thdgustjd1**
# 📝 PR 타입
- [x] 기능 구현
- [x] 기능 수정
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 인프라, 의존성, 환경 변수, 빌드 관련 코드 업데이트

# 🚀 변경 사항
##  Stream을 이용한 Route 조회 개선
- 기존에는 List<Route>로 전체 데이터를 한 번에 조회하다 보니, 한정된 EC2 인스턴스 리소스 환경에서 수만 개의 Node와 Route 데이터를 여러 쓰레드가 동시에 조회할 때 OOM(OutOfMemory) 문제가 발생했습니다.
- 이를 해결하기 위해, 전체 데이터를 한꺼번에 메모리에 적재하는 대신, Stream<Route>를 활용하여 일정 단위(예: 2500건)로 데이터를 처리하도록 변경했습니다.
- 각 배치 처리 후 entityManager.detach()와 리스트 클리어 작업을 통해 영속성 컨텍스트 및 메모리에서 불필요한 데이터를 제거하여, GC가 효과적으로 메모리를 회수할 수 있도록 구현하였습니다.

## SSE를 통한 분할 응답
- 전체 데이터를 한 번의 응답으로 전달하면 응답 데이터가 누적되어 메모리 사용량이 급증할 수 있는 문제가 있었습니다.
- 이를 개선하기 위해, Server-Sent Events(SSE)를 도입하여 데이터를 배치 단위로 분할하여 전송하도록 변경하였습니다.
- 이 방식은 클라이언트가 각 배치 데이터를 즉시 처리할 수 있게 하여, 서버 측에서는 응답 데이터를 누적하지 않고 지속적으로 메모리를 해제할 수 있도록 합니다.
- 또한, Stream과 SSE 기법을 결합하여 메모리 부담을 최소화하면서 응답 속도와 처리 효율을 극대화하였습니다.

## 힙 사용량 비교자료
조건 : 힙 메모리 500MB, 쓰레드 15개, 루프카운트 10

image
image

# 🧪 테스트 결과

image

[UNI-324]: https://uniro.atlassian.net/browse/UNI-324?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mikekks — APPROVED** [review]
제안해주신 스트림 방식 좋은 것 같습니다! 훨씬 더 안정적인 메모리 운용이 가능할 것 같아요! 고생하셨습니다.

File: uniro_backend/src/main/java/com/softeer5/uniro_backend/common/config/AsyncConfig.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+@Configuration
+public class AsyncConfig {
+
+    @Bean
+    public ThreadPoolTaskExecutor taskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(10); //기본 스레드 개수 10개
+        executor.setMaxPoolSize(20); //20개까지 추가 생성 가능
+        executor.setQueueCapacity(100); //대기열 크기
```
[the remark this answers is not in the document]
Comment (thdgustjd1) [reply uniro_backend/src/main/java/com/softeer5/uniro_backend/common/config/AsyncConfig.java:17]: 해당 코드들은 전부 비동기 프로그래밍을 위해 대기중인 스레드의 개수들인데, 여러 테스트들과 캐시가 곧 사용될 것을 고려하여 세팅해둔 값입니다!
